src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -74,6 +102,208 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+            lineItems,
+        }];
+
+        const checkout = await this._paypalCommercePaymentProcessor.getShippingOptions(cart.id, payload);
+        const availableShippingOptions = (checkout as CheckoutWithBillingAddress).consignments[0].availableShippingOptions;
+        const shippingRequired = (checkout as CheckoutWithBillingAddress ).cart.lineItems.physicalItems.length > 0;
+        if (!shippingRequired) {
+            const patch = await actions.order.patch([
+                {
+                    op: 'replace',
+                    path: '/purchase_units/@reference_id==\'default\'/amount',
+                    value: {
+                        currency_code: 'USD',
+                        value: (parseFloat(String(baseOrderAmount))).toFixed(2),
+                        breakdown: {
+                            item_total: {
+                                currency_code: 'USD',
+                                value: baseOrderAmount,
+                            },
+                        },
+                    },
+                },
+            ]);
+
+            return patch;
+            // If no shipping options returned, but shipping is required, do not allow to submit such order
+        } else if (shippingRequired && availableShippingOptions?.length === 0) {
+            return actions.reject();
+        } else {
+            const shippingOptions: ShippingOptions[] = [];
+            availableShippingOptions?.forEach((option: AvaliableShippingOption) => {
+                let isSelected = false;
+                // Buyer has chosen shipping option on PP list and address the same
+                if (data.selected_shipping_option && this._isAddressSame(
+                    this._currentShippingAddress, this._submittedShippingAddress
+                )) {
+                    if (option.id === data.selected_shipping_option.id) {
+                        shippingAmount = data.selected_shipping_option.amount.value;
+                        isSelected = true;
+                    }
+                } else {
+                    if (option.isRecommended) {
+                        shippingAmount = parseFloat(String(option.cost)).toFixed(2);
+                        isSelected = true;
+                    }
+                }
+                shippingOptions.push({
+                    id: option.id,
+                    type: 'SHIPPING',
+                    label: option.description,
+                    selected: isSelected,
+                    amount: {
+                        value: parseFloat(String(option.cost)).toFixed(2),
+                        currency_code: 'USD',
+                    },
+                });
+            });
+
+            shippingOptions.sort( (a, b) => {
+                // @ts-ignore
+                return b.selected - a.selected;
+            });
+            if (shippingOptions && shippingOptions[0].id) {
+                this._shippingOptionId = shippingOptions[0].id;
+            }
```

Comment:
Do customer have an ability to select multiple shipping options?

Reply:
No, just one single option

Reply:
In this case we can make it much simpler I think.